src/components/input-message/input-message.tsx

```diff
@@ -35,6 +35,9 @@ export class InputMessage {
   /** Specifies the status of the input field, which determines message and icons. */
   @Prop({ reflect: true, mutable: true }) status: Status = "idle";
 
+  /** Specifies if the message should be visible to the user */
+  @Prop({ reflect: true, mutable: true }) hidden = false;
```

Comment:
Hey @alisonailea, instead of adding a `hidden` property, can you create a container `<div>` inside the component  that surrounds all the internal elements and hide it based on the textContent logic? The reasoning is that there's already a global `hidden` attribute on HTMLElement's and we don't want to override any `hidden` attribute or `display:none` that a user sets on the host element. 

For example, if a user added `hidden` to this component it would be removed if it had textContent.

It's been a best practice that we don't change the display of an element during its lifecycle. I think I can add this to the conventions if its not already there.

Proposal:

```jsx
<Host calcite-hydrated-hidden={hidden}>
    <div class={{["has-content"]: hasContent}}>
        {this.renderIcon(this.requestedIcon)}
        <slot />
    </div>
</Host>
```